sdk/src/cameras/itof-frame/frame_handler_impl.cpp

```diff
@@ -115,13 +115,55 @@ Status FrameHandlerImpl::saveFrameToFile(aditof::Frame &frame,
 
     m_file.close();
 
+    if (!m_frameQueue.empty()) {
+        m_mutex.lock();
+        m_frameQueue.pop();
+        m_frameNameQueue.pop();
+        m_mutex.unlock();
+    }
+
     return status;
 }
 
-Status FrameHandlerImpl::saveFrameToFileMultithread(aditof::Frame *frame,
+Status FrameHandlerImpl::saveFrameToFileMultithread(aditof::Frame &frame,
                                                     std::string fileName) {
 
-    return aditof::Status::UNAVAILABLE;
+    using namespace aditof;
+    Status status = Status::OK;
+
+    m_mutex.lock();
+    m_frameQueue.push(std::move(frame));
+    m_frameNameQueue.push(fileName);
+    m_mutex.unlock();
+
+    frame = Frame();
+
+    if (!m_threadRunning) {
+        if (m_threadWorker.joinable()) {
+            m_threadWorker.join();
+        }
+
+        m_threadWorker =
+            std::thread(std::bind(&FrameHandlerImpl::threadWritter, this));
+        m_threadRunning = true;
```

Comment:
you are already setting this in threadWritter routine

Reply:
Fixed, thank you!